src/main/kotlin/interactor/GetTop10CountriesEnforceHighTaxesOnCarbonatedDrinks.kt

```diff
@@ -0,0 +1,41 @@
+package interactor
+
+import model.CityEntity
+
+class GetTop10CountriesEnforceHighTaxesOnCarbonatedDrinks (
+    private val dataSource: CostOfLivingDataSource,
+){
+
+    fun excute():List<String>?{
+        val list = dataSource.getAllCitiesData()
+
+        if (list.isEmpty()||allPriceIsNull(list)||allDataIsLow(list)) return null
+
+        return dataSource.getAllCitiesData()
+            .filter(::excludeNullPricesAndLowQualityData)
+            .sortedByDescending{it.drinksPrices.cokePepsiAThirdOfLiterBottleInRestaurants}
+            .take(10).map { it.cityName +" : " + it.drinksPrices.cokePepsiAThirdOfLiterBottleInRestaurants}
+
+    }
+    private fun excludeNullPricesAndLowQualityData(city: CityEntity): Boolean {
+        return city.drinksPrices.cokePepsiAThirdOfLiterBottleInRestaurants != null
+                && city.dataQuality
+    }
+    private fun allPriceIsNull(list: List<CityEntity>):Boolean{
```

Comment:
Why not using declarative approach instead?
see [all()](https://kotlinlang.org/docs/collection-filtering.html#test-predicates)